librosa/display.py

```diff
@@ -859,6 +867,9 @@ def specshow(
     __decorate_axis(axes.xaxis, x_axis, key=key, Sa=Sa, mela=mela, thaat=thaat)
     __decorate_axis(axes.yaxis, y_axis, key=key, Sa=Sa, mela=mela, thaat=thaat)
 
+    # If the plot is a self-similarity/covariance etc. plot, square it
+    if __same_axes(x_axis, y_axis, axes.get_xlim(), axes.get_ylim()) and auto_scale:
+        axes.set_aspect('equal')
```

Comment:
Do we only want to apply this when the axes are the same type?  There are cases where we have distinct scales (`cqt_note` vs `cqt_hz` for example) that cover the same units and data extents.

I guess this is a bit of an edge case, and can always be handled explicitly after displaying by manually setting the aspect ratio, but it's worth considering and/or documenting.

Reply:
I think we can punt on this -- better to keep it simple and predictable.